connectors/src/connectors/google_drive/temporal/file.ts

```diff
@@ -324,6 +328,49 @@ export async function syncOneFile(
               return false;
             }
           }
+        } else if (file.mimeType === "text/csv") {
+          if (res.data instanceof ArrayBuffer) {
+            // If data is > 4 times the limit, we skip the file since even if
+            // converted to utf-8 it will overcome the limit enforced below. This
+            // avoids operations on very long text files, that can cause
+            // Buffer.toString to crash if the file is > 500MB
+            if (res.data.byteLength > 4 * maxDocumentLen) {
+              localLogger.info({}, `File too big to be chunked. Skipping`);
+              return false;
+            }
+            const tableCsv = Buffer.from(res.data).toString("utf-8").trim();
+            const tableId = file.id;
+            const tableName = slugify(file.name.substring(0, 32));
+            const tableDescription = `Structured data from Google Drive (${file.name})`;
+
+            await upsertTableFromCsv({
```

Comment:
You should check the result here, as it could fails if the CSV is not "valid" (missing headers, invalid number of columns, ...)

Reply:
✅ Took inspiration from other places in the codebase